### ISSUE TITLE
优化 CPUAttentionGrad.cpp 的反向计算性能

### ISSUE THREAD

**Issue by leeyy2020** [leeyy2020/MNNold#3]
## 优化目标
提升 CPUAttentionGrad.cpp 的性能，特别是 Q @ K^T 点积部分，避免每个 i,j 的逐元素循环，改为批量 packed-matmul 实现。参考 CPUAttention.cpp 的前向优化方法，将 K（和后续可选的 V）做一次 pack 后复用。这样可以显著加速 logits 的计算以及减少内存分配和拷贝。

## 主要修改方案
- 用 core->MNNPackForMatMul_B 对 K 做 pack，仅一次，后续复用。
- logits（Q * K^T * scale）用 core->MNNPackedMatMul / MNNPackedMatMulRemain 批量计算，避免内层循环。
- K、V 在每个 head 上只 copy/pack 一次。
- 保持 softmax、softmax 反向和梯度累加逻辑不变，保证数值等价。
- 保留多线程并行，所有临时 buffer 线程独立分配。

## 进一步优化建议
- 后续可用 packed-matmul 优化 dV = P^T * dY、dQ = (dS * scale) * K、dK = dS^T * Q 等所有矩阵乘法。
- 避免多次 memcpy，可以增加 stride/offset 支持。
- 如果还有性能瓶颈，可考虑更底层的 SIMD 优化。

## 代码片段（伪代码）
```cpp
// pack K
core->MNNPackForMatMul_B(packB_K.data(), Khead.data(), mHeadDim, mKvSeq, true);
// logits = Q @ K^T
for (i块) {
  core->MNNPackedMatMul(Cout, Ablock, packB_K.data(), ...);
}
```

## 预期收益
- QK 部分提速数倍，整体反向性能提升明显（尤其 head、seq 较大时）。
- 保证与原算法数值一致。

## 影响范围
- 仅影响 CPUAttentionGrad.cpp 的反向实现，不涉及前向、量化、KVCache 逻辑。
- 仅支持 float32 路径（FP16/INT8 需后续适配）。

---
如需进一步批量优化 dQ, dK, dV 的矩阵乘法，请补充需求。
